experimental/dds/tree2/src/test/shared-tree/sharedTreeView.spec.ts

```diff
@@ -0,0 +1,70 @@
+/*!
+ * Copyright (c) Microsoft Corporation and contributors. All rights reserved.
+ * Licensed under the MIT License.
+ */
+import { strict as assert } from "assert";
+import { SchemaBuilder, Any } from "../../feature-libraries";
+import { createSharedTreeView } from "../../shared-tree";
+import { ValueSchema, AllowedUpdateType, storedEmptyFieldSchema } from "../../core";
+
+const builder = new SchemaBuilder("Schematize Tree Tests");
+const root = builder.leaf("root", ValueSchema.Number);
+const schema = builder.intoDocumentSchema(SchemaBuilder.fieldOptional(Any));
+
+const builderGeneralized = new SchemaBuilder("Schematize Tree Tests Generalized");
+const rootGeneralized = builderGeneralized.leaf("root", ValueSchema.Serializable);
+const schemaGeneralized = builderGeneralized.intoDocumentSchema(SchemaBuilder.fieldOptional(Any));
+
+describe("sharedTreeView", () => {
+	describe("schematize", () => {
+		it("initialize tree", () => {
+			const tree = createSharedTreeView();
+			assert.equal(tree.storedSchema.rootFieldSchema, storedEmptyFieldSchema);
+
+			tree.schematize({
+				allowedSchemaModifications: AllowedUpdateType.None,
+				initialTree: 10 as any,
+				schema,
+			});
+			assert.equal(tree.root, 10);
+		});
+
+		it("noop upgrade", () => {
+			const tree = createSharedTreeView();
+			tree.storedSchema.update(schema);
+
+			// No op upgrade with AllowedUpdateType.None does not error
+			const schematized = tree.schematize({
+				allowedSchemaModifications: AllowedUpdateType.None,
+				initialTree: 10,
+				schema,
+			});
+			// And does not add initial tree:
+			assert.equal(schematized.root, undefined);
+		});
+
+		it("incompatible upgrade errors", () => {
```

Comment:
super nit: the tests say "upgrade" but the method is "update", I don't care if you keep it like this but it's kinda funny.

Reply:
the update method on storedSchema is not whats being tested here. Its jusr being used to initialize the state which schematize is going to fail to upgrade, and is unrelated to the name of this test.

More specifically, storedSchema.update can be used to do changes that are downgrades (ex: undoing an upgrade), merge resolutions etc, so it being calles something less speciifc than upgrade makes sense. Schematize is monotonic, and will error if it does something that doesn't look like a version upgrade.